tests/harness/go/proto_names/harness.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"google.golang.org/protobuf/reflect/protoreflect"
+
+	_ "github.com/envoyproxy/protoc-gen-validate/tests/harness/cases/go_proto_field_name"
+	_ "github.com/envoyproxy/protoc-gen-validate/tests/harness/cases/other_package/go_proto_field_name"
+	harness "github.com/envoyproxy/protoc-gen-validate/tests/harness/go"
+)
+
+func main() {
+	b, err := ioutil.ReadAll(os.Stdin)
+	checkErr(err)
+
+	tc := new(harness.TestCase)
+	checkErr(proto.Unmarshal(b, tc))
+
+	da := new(ptypes.DynamicAny)
+	checkErr(ptypes.UnmarshalAny(tc.Message, da))
+	v := da.Message.(interface {
+		Validate() error
+	})
+
+	checkMsg(da.Message, v.Validate())
+}
+
+func checkMsg(message proto.Message, err error) {
+	if err == nil {
+		resp(&harness.TestResult{Valid: true}) // no error means no error message to be checked
+	}
+	fieldError, ok := err.(interface{ Field() string })
+	if !ok {
+		resp(&harness.TestResult{Error: true,
+			Reason: fmt.Sprintf("error does not implement the `interface{ Field() string }`")})
+	}
+	field := fieldError.Field()
+	if field == "" {
+		resp(&harness.TestResult{Error: true,
+			Reason: fmt.Sprintf("error does not contain the field")})
+	}
+	msg := proto.MessageReflect(message)
+	fieldNames := listAllFields(msg)
+	if !contains(field, fieldNames) {
+		resp(&harness.TestResult{Error: true,
+			Reason: fmt.Sprintf("error message '%s' does not contain the field '%s'", err.Error(), field)})
+	}
+	resp(&harness.TestResult{Reason: err.Error()})
+}
```

Comment:
Why is this implemented in a separate harness instead of the existing one?

Reply:
I tried to implement the checks in the same harness but failed because proto field names require a separate protoc code generation of the testcases. This means I cannot import both corresponding packages in `tests/harness/go/main/harness.go`. If you would like such an error message test for the normal harness as well, I could try to refactor the code so that both harnesses check the field naming in the error message. Having only one test harness requires a more invasive change in the test design as the testcase protoc code generation needs to be triggered multiple times before the harness.